### PR TITLE
fix a colspan issue while using the detailformatter

### LINKED ISSUE #5184: 1.17.1版本 Bug
**Bootstraptable version(s) affected**:

**Description**  

启用 detailView，并添加操作列，当数据为空时，表格显示错位，详细如图
1.16.0 版本不存在该问题
**Example**  

image

**Possible  (optional)**  

**Additional context**

### PR DESCRIPTION
**Pull request by UtechtDustin**
**Bug fix?**
yes

**New Feature?**
no

**Resolve an issue?**
Fix #5184 

**Example(s)?**
Before: https://live.bootstrap-table.com/code/UtechtDustin/4016
After: https://live.bootstrap-table.com/code/UtechtDustin/4017

### CODE REVIEW

File: src/bootstrap-table.js [hunk 1]
```diff
@@ -1538,7 +1538,7 @@ class BootstrapTable {
     // show no records
     if (!hasTr) {
       this.$body.html(`<tr class="no-records-found">${Utils.sprintf('<td colspan="%s">%s</td>',
-        this.getVisibleFields().length,
+        this.getVisibleColumnCount(),
```
Comment (wenzhixin) [inline src/bootstrap-table.js:1541]:
```
this.getVisibleFields().length +
  (Utils.hasDetailViewIcon(this.options) && this.options.detailViewAlign !== 'right' ? 1 : 0)

Comment (wenzhixin) [inline src/bootstrap-table.js:1541]:
```
this.getVisibleFields().length +
  (Utils.hasDetailViewIcon(this.options) && this.options.detailViewAlign !== 'right' ? 1 : 0)

**Comment by wenzhixin** [comment]
We can use `Utils.hasDetailViewIcon` method.

File: src/bootstrap-table.js (hunk 1 above)
Comment (UtechtDustin) [reply src/bootstrap-table.js:1541]: I agree we can use the Utils.hasDetailViewIcon method, but we should use a function as wrapper for that.
If we have in the future new features which must be also checked, the code will be ugly.
So we can prevent it right now.

What do you think @wenzhixin

Comment (wenzhixin) [reply src/bootstrap-table.js:1541]: OK, let me updated this PR.

**Comment by wenzhixin** [comment]
Updated and fixed a new problem with `mergeCells`.

Before: https://live.bootstrap-table.com/code/wenzhixin/4125
After: https://live.bootstrap-table.com/code/wenzhixin/4126

**Comment by UtechtDustin** [comment]
Looks good for me, good job.

**Pull request merged**